theano/tensor/tests/test_mpi.py

```diff
@@ -46,17 +46,18 @@ def test_mpi_roundtrip():
     if not mpi_enabled:
         raise SkipTest('MPI not enabled')
     theano_root = theano.__file__.split('__init__')[0]
-    d = {}
-    if PY3:
-        # Is some not understood cases, the subprocess never finish.
-        d = dict(timeout=5*60)
+    env = os.environ.copy()
+    flags = env['THEANO_FLAGS']
+    keep_flags = ','.join((f for f in flags.split(',') if not f.startswith('init_gpu_device')))
```

Comment:
Also filter out device?

Reply:
For the tests device is supposed to be 'cpu' so that's why I didn't filter it out.